src/main/healthCheck.ts

```diff
@@ -0,0 +1,50 @@
+import log from 'electron-log/main'
+import { HttpsProxyAgent } from 'https-proxy-agent'
+import https from 'node:https'
+
+import { getProxyCertificateContent, getProxyURL } from './proxy'
+
+export const checkProxyHealth = async () => {
+  try {
+    const results = await Promise.allSettled([
+      isUrlReachable('https://www.google.com/generate_204'),
+      isUrlReachable('https://quickpizza.grafana.com'),
+      isUrlReachable('https://k6.io'),
+    ])
+    const isProxyHealthy = results.some(
+      (r) => r.status === 'fulfilled' && r.value === true
+    )
+    return isProxyHealthy
+  } catch (error) {
+    log.error('Error checking proxy health:', error)
+    return false
+  }
+}
+
+const isUrlReachable = async (url: string) => {
+  return new Promise((resolve) => {
+    const certContent = getProxyCertificateContent()
+    const agent = new HttpsProxyAgent(getProxyURL())
+    const options: https.RequestOptions = {
+      agent,
+      ca: certContent,
+      headers: {
+        'k6-Studio-Health-Check': 'true',
```

Comment:
```suggestion
        'X-K6-Studio-Health-Check': 'true',
```
Since the request will be hitting the server, it might be nice to follow convention.